cpu/cc3200/periph/gpio.c

```diff
@@ -0,0 +1,393 @@
+    }
+
+    /* set strength and drain mode */
+    gpio_init_af(dev, PIN_STRENGTH_2MA, mode & PAD_TYPE_MASK);
+
+    return 0;
+}
+
+#ifdef MODULE_PERIPH_GPIO_IRQ
+
+void isr_gpio_a0(void)
+{
+    handle_isr((cc3200_gpio_t *)GPIOA0_BASE);
+}
+
+void isr_gpio_a1(void)
+{
+    handle_isr((cc3200_gpio_t *)GPIOA1_BASE);
+}
+
+void isr_gpio_a2(void)
+{
+    handle_isr((cc3200_gpio_t *)GPIOA2_BASE);
+}
+
+void isr_gpio_a3(void)
+{
+    handle_isr((cc3200_gpio_t *)GPIOA3_BASE);
+}
+
+/**
+ * @brief isr interrupt handler
+ *
+ * @param portAddr base address of the GPIO PORT
+ */
+void handle_isr(cc3200_gpio_t *u)
+{
+    uint32_t state = u->mis;
+
+    /* clear interrupt */
+    u->icr = state;
+    for (int i = 0; i < 8; i++) {
+        if (state & (1 << i)) {
+            isr_ctx[port_num][i].cb(isr_ctx[port_num][i].arg);
+        }
+    }
+
+    cortexm_isr_end();
+}
+
+/**
+ * @brief enable GPIO interrupt
+ * @param dev pin
+ */
+void gpio_irq_enable(gpio_t dev)
+{
+    gpio(dev)->im |= gpio_pin_mask(dev);
+}
+
+/**
+ * @brief disable GPIO interrupt
+ * @param dev pin
+ */
+void gpio_irq_disable(gpio_t dev)
+{
+    gpio(dev)->im &= ~(gpio_pin_mask(dev));
+}
+
+int gpio_init_int(gpio_t dev, gpio_mode_t mode, gpio_flank_t flank,
+                  gpio_cb_t cb, void *arg)
+{
+    /* Note: gpio_init() also checks if the gpio is enabled. */
+    int res = gpio_init(dev, mode);
+    if (res != 0) {
+        return res;
+    }
+
+    assert(flank != GPIO_NONE);
+
+    uint8_t portNum  = gpio_port_num(dev);
+    uint8_t pinNum   = gpio_pin_num(dev);
+    uint8_t bit      = gpio_pin_mask(dev);
+    uint8_t portAddr = gpio_pin_to_port(dev);
+
+    /* store callback information; */
+    isr_ctx[portNum][pinNum].cb  = cb;
+    isr_ctx[portNum][pinNum].arg = arg;
+
+    ROM_IntMasterDisable();
+
+    /* clear interrupt specified pin */
+    gpio(dev)->ICR = bit;
+
+    /* configure active flanks */
+    gpio(dev)->ibe =
+            (flank & GPIO_BOTH) ? gpio(dev)->ibe | bit : gpio(dev)->ibe & ~bit;
+    gpio(dev)->is = (flank & GPIO_VALUE_LOW) ? gpio(dev)->is | bit :
+                                               gpio(dev)->is & ~bit;
+    gpio(dev)->iev = (flank & GPIO_RISING) ? gpio(dev)->iev | bit :
+                                             gpio(dev)->iev & ~bit;
+
+    /* enable gpio interripts */
+    gpio(dev)->im |= bit;
+
+    /* register interrupt handlers */
+    /* TODO: replace with cortex common */
+    switch (portBase) {
+    case GPIOA0_BASE:
+        ROM_GPIOIntRegister(portBase, isr_gpio_a0);
+        ROM_IntEnable(INT_GPIOA0);
+        break;
+    case GPIOA1_BASE:
+        ROM_GPIOIntRegister(portBase, isr_gpio_a1);
+        ROM_IntEnable(INT_GPIOA1);
+        break;
+    case GPIOA2_BASE:
+        ROM_GPIOIntRegister(portBase, isr_gpio_a2);
+        ROM_IntEnable(INT_GPIOA2);
+        break;
+    case GPIOA3_BASE:
+        ROM_GPIOIntRegister(portBase, isr_gpio_a3);
+        ROM_IntEnable(INT_GPIOA3);
+        break;
```

Comment:
Well the reference manual only mentions port 0 till 3 and as described here some pins require pretty [special configuration](http://e2e.ti.com/support/wireless-connectivity/wifi/f/968/t/386871) to be used as GPIOs. I can add port 4 but don't think anyone would or should use it since its pretty sketchy.

Reply:
as mentioned above I have included it for completeness sake but I would still add some disclaimer to the board description about the limitations of port A4